allennlp/modules/drop_connect.py

```diff
@@ -75,6 +77,18 @@ def _search_parameters(self, regex: str) -> Iterator[_Match]:
                 if re.search(regex, parameter_name):
                     yield _Match(submodule_name, submodule, parameter_name, parameter)
 
+    def _prepare_matched_parameters(self, match: _Match):
+        """
+        Apply non-training dropout to all of matched parameters otherwise PyTorch will raise
+        `AttributeError`: '`LSTM`' object has no attribute '`_flat_weights`'.
+        The `forward()` method, however, will still warn against fragmented memory usages of
+        non-flatten parameters. A corresponding workaround is in `forward()`.
+        """
+        raw_parameter = getattr(self, match.raw_parameter_name)
+        match.module._parameters[match.parameter_name] = F.dropout(
```

Comment:
Why apply dropout with training=False instead of just assigning the parameter?

Reply:
As https://github.com/fastai/fastai/commit/b842586e9b080ed83afb251d4236ec6843d823de described, it is probably a workaround for and only for QRNN with discriminative fine-tuning. That was the main reason why I was reluctant to submit this PR. Another related concern is that cloning the weight by the identity function of dropout somewhat contradicts the intention of `delattr()`. By the way, while `delattr()` seems to destroy the `all_weights`, the original Salesforce version uses a less convenient yet seemingly more surgical way of `del self.module._parameters[name_w]`.